guides/web-apps/acceptance-testing-with-selenium/README.md

```diff
@@ -114,157 +84,116 @@ We don't want to test only Firefox. Because of that, you should use Selenium Rem
-                break;
-            case "edge":
-                capabilities = DesiredCapabilities.Edge();
-                driver = new RemoteWebDriver(remoteWebDriverUri, capabilities);
-                break;
-            default:
-                capabilities = DesiredCapabilities.Firefox();
-                driver = new RemoteWebDriver(remoteWebDriverUri, capabilities);
-                break;
-        }
-        return driver;
-    }
-}
-```
-
-Now, we need to adapt the test file to make use of it.
+When you rebuild the test project now, you should see each test for every browser.
 
-In the file you created in the previous step (`PageLoads.cs`), locate the following line:
+The final setup looks like this:
 
-```cs
-driver = new FirefoxDriver();
-```
+![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
 
-Change it to:
+Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.*.jar`.
 
-```cs
-driver = WebDriverFactory.Create(this.browser);
-```
+## Wait for asynchronous content
 
-Now, what is `this.browser`? This is a fixture parameter that carries the name of the browser used to execute the test.
+There is one common pitfall when writing Selenium tests. The test is executed with disregard of the asynchronously loaded content. This means that your tests need to explicitly wait for UI elements to appear before running any actions or assertions on them.
 
-What this means is that you need to change every instance of :
+It is a good practice to always wait:
 
+- Wait for a text element to be present before you check the content of that element
 ```cs
-[TestFixture]
+public bool WaitForText(IWebElement elementName, string text, int seconds)
+{
+	WebDriverWait wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(seconds));
+	return wait.Until(ExpectedConditions.TextToBePresentInElement(elementName, text));
+}
 ```
+Using:
+```cs
+[Test]
+public void ButtonPage_RegularButton()
+{
+	WaitUntil(x => _radiolistPage.InfoLabel.Displayed);
+	Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Dogs", 5));
 
-To be:
+	_radiolistPage.SelectRadio("Cats");
+	Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Cats", 5));
 
+	_radiolistPage.SelectRadio("Dogs");
+	Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Dogs", 5));
+}
+```
+- Wait for a button to be present before you click on that button
 ```cs
-[TestFixture("firefox")]
-[TestFixture("chrome")]
-[TestFixture("edge")]
+public IWebElement WaitForElementToBeClickable(IWebElement elementName, int seconds)
+{
+	WebDriverWait wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(seconds));
+	return wait.Until(ExpectedConditions.ElementToBeClickable(elementName));
+}
+public void ClickOn(IWebElement elementName, int seconds = 10)
+{
+	IWebElement element = WaitForElementToBeClickable(elementName, seconds);
+	element.Click();
+}
```

Comment:
This is not a usage example because it is out of context:

```cs
public void ClickButtonInlineScript()
{
	ClickOn(ButtonInlineScript);
}

public void ClickButtonFunction()
{
	ClickOn(ButtonFunction);
}
```

This is a usage example:

```
[Test]
        public void ButtonPage_RegularButton()
        {
            WaitUntil(x => _radiolistPage.InfoLabel.Displayed);
            Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Dogs", 5));

            _radiolistPage.SelectRadio("Cats");
            Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Cats", 5));

            _radiolistPage.SelectRadio("Dogs");
            Assert.IsTrue(WaitForText(_radiolistPage.InfoLabel, "Dogs", 5));
        }
```

At this point I agree with @Mackiovello that the snippets are not the good way to go and they should be changed to links. In the previous version, a 15-line snippet showed everything that was required. Now there are more snippets but still they don't show 50% of the code involved, because much logic is moved to the Page classes.